lib/server/connect.js

```diff
@@ -10,7 +22,92 @@ SubHandlers = {
 var uniqueId = 0;
 var serverId = Random.id();
 
-Meteor.publish('kadira.debug.init', function(browserId, clientId) {
+// publications
+
+Meteor.publish('kadira.debug.remote.auth', function(authKey) {
+  check(authKey, Match.Any);
+
+  if(AppConfig.env === 'production') {
+    if(authKey !== AppConfig.authKey) {
+      throw new Meteor.Error('401', 'Unauthorized.');
+    }
+
+    var kadiraInfo = Kadira._getInfo();
+    var sessionId = kadiraInfo.session;
+
+    AppConfig._authorizedSessions.remote[sessionId] = true;
+  }
+
+  this.onStop(function() {
+    delete AppConfig._authorizedSessions.remote[sessionId];
+  });
+
+  this.ready();
+});
+
+Meteor.publish('kadira.debug.remote.timeline', function() {
+
+  var kadiraInfo = Kadira._getInfo();
+
+  var isAuthorized = isAuthorizedSession('remote', kadiraInfo);
+  if(!isAuthorized) {
```

Comment:
You don't need to check it here. 
If not authorized, isAuthorizedSession will throw the error.